src/components/vectorfeedback/VectorFeedbackDirective.js

```diff
@@ -0,0 +1,157 @@
+      gaMapUtils,
+      gaGLStyle,
+      gaBackground
+  ) {
+    return {
+      restrict: 'A',
+      replace: true,
+      templateUrl: 'components/vectorfeedback/partials/vectorfeedback.html',
+      scope: {
+        map: '=gaVectorFeedbackMap',
+        options: '=gaVectorFeedbackOptions',
+        submit: '=gaVectorFeedbackSubmit'
+      },
+      link: function(scope, element, attrs) {
+        var map = scope.map;
+        var mobile = scope.options.mobile;
+
+        // Change the color
+        var applyColor = function(color) {
+          if (!mobile) {
+            scope.options.activeColor = color;
+          }
+          var olLayers = gaMapUtils.getMapBackgroundLayersArray(map);
+          var edit = scope.options.selectedLayer.edit.slice();
+          edit[2] = edit[2].replace('{color}', color);
+          var glStyle = gaGLStyle.edit([edit]);
+          gaMapUtils.applyGLStyleToOlLayers(
+              olLayers,
+              glStyle.style,
+              glStyle.sprite
+          );
+        };
+
+        var registerSelectedLayerWatcher = function() {
+          return scope.$watch('options.selectedLayer',
+              function(newVal, oldVal) {
+                if (newVal && oldVal && newVal.value !== oldVal.value) {
+                  scope.options.activeColor = '';
+                  var olLayers = gaMapUtils.getMapBackgroundLayersArray(
+                      map);
+                  var glStyle = gaGLStyle.resetEdits();
+                  gaMapUtils.applyGLStyleToOlLayers(
+                      olLayers,
+                      glStyle.style,
+                      glStyle.sprite
+                  );
+                }
+              });
+        };
+
+        var registerBackgroundLayerWatcher = function() {
+          return scope.$watch('options.backgroundLayer', function(newVal) {
+            var olLayer = gaMapUtils.getMapBackgroundLayer(map);
+            // Dropdown interaction
+            if (olLayer && olLayer.bodId !== newVal.id) {
+              gaBackground.setById(map, newVal.id);
+            }
+          });
+        };
+
+        var registerShowLabelWatcher = function() {
+          return scope.$watch('options.showLabel', function(newVal, oldVal) {
+            if (newVal && oldVal && newVal.value !== oldVal.value) {
+              var glStyle;
+              var olLayers = gaMapUtils.getMapBackgroundLayersArray(map);
+              if (olLayers) {
+                if (newVal.value) {
+                  glStyle = gaGLStyle.resetFilters();
+                } else {
+                  var bodId = scope.options.backgroundLayer.id;
+                  var filters =
+                    scope.options.layers[bodId].labelsFilters;
+                  if (filters) {
+                    glStyle = gaGLStyle.filter(filters);
+                  }
+                }
+                if (glStyle && glStyle.style) {
+                  gaMapUtils.applyGLStyleToOlLayers(
```

Comment:
Can t we just apply the style the Layer group ?

Reply:
I followed what has been done in `gaLayers`, since you assign a style function to each child layer I need to override them.

Reply:
@oterral if it works by putting it at the group level it would be much better but I am not sure.
I would address this in a separate PR though.
Let's merge and do that later, changes here will be trivial.

Reply:
Yes but in `gaLayers` I have to follow what was done before for the 'aggregate ' . I think an applyGLStyleToOlLayer which manages ol.layer.Group should be easire to use.
I change it, it will be fixed when you wake up.